q2cli/cli.py

```diff
@@ -0,0 +1,104 @@
+                   'official QIIME 2 plugins at: https://github.com/qiime2.')
+    else:
+        click.echo('Installed plugins:')
+        for name, plugin in installed_plugins:
+            click.echo(' %s %s (%s)' %
+                       (name, plugin.version, plugin.website))
+
+# should the QIIME types be hashable to support this without casting
+# to a string?
+_type_map = {'Int': click.INT, 'Str': click.STRING, 'Float': click.FLOAT}
+
+def build_commands(group):
+    plugin_manager = PluginManager()
+    for plugin_name, plugin in plugin_manager.plugins.items():
+        for workflow_name, workflow in plugin.workflows.items():
+            parameters = []
+            for ia_name, ia_type in workflow.signature.input_artifacts.items():
+                p = click.Option(['--%s' % ia_name],
+                                   required=True,
+                                   type=click.Path(exists=True, dir_okay=False),
+                                   help='Input %s' % str(ia_type))
+                parameters.append(p)
+            for ip_name, ip_type in workflow.signature.input_parameters.items():
+                p = click.Option(['--%s' % ip_name],
+                                   required=True,
+                                   type=_type_map[str(ip_type)],
+                                   help=str(ip_type))
+                parameters.append(p)
+            for oa_name, oa_type in workflow.signature.output_artifacts.items():
+                p = click.Option(['--%s' % oa_name],
+                                   required=True,
+                                   type=click.Path(exists=False, dir_okay=False),
+                                   help='Output %s' % str(oa_type))
+                parameters.append(p)
+
+            def create_callback(wf):
+                def f(**kwargs):
+                    # execute workflow
+                    executor = SubprocessExecutor()
+                    input_artifacts = {
+                        ia_name: kwargs[ia_name]
+                        for ia_name in wf.signature.input_artifacts.keys()}
+                    input_parameters = {
+                        ip_name: kwargs[ip_name]
+                        for ip_name in wf.signature.input_parameters.keys()}
+                    output_artifacts = {
+                        oa_name: kwargs[oa_name]
+                        for oa_name in wf.signature.output_artifacts.keys()
+                    }
+                    future_ = executor(wf,
+                                       input_artifacts,
+                                       input_parameters,
+                                       output_artifacts)
+
+                    # if in the future we want the cli to not block but rather
+                    # start daemonized processes, we just don't access result
+                    # here, as that's the step that blocks
+                    completed_process = future_.result()
```

Comment:
Would it be possible to have a `--daemonize` exposed at the plugin or higher level?